lib/collection/src/update_handler.rs

```diff
@@ -62,11 +62,12 @@ pub struct UpdateHandler {
     runtime_handle: Handle,
     /// WAL, required for operations
     wal: Arc<Mutex<SerdeWal<CollectionUpdateOperations>>>,
+    blocking_handles: Arc<Mutex<Vec<JoinHandle<()>>>>,
```

Comment:
nit: I would prefer to call it `optimization_handles` instead of `blocking_handles`, as it better describes semantics of the variable. Also the information that it is blocking is already presents in the type.